src/server.rs

```diff
@@ -293,6 +294,20 @@ async fn notify_device(
     }
 
     info!("Got direct notification for {device_token}.");
+    let now = Instant::now();
+    if state.debouncer().is_debounced(now, &device_token) {
```

Comment:
To avoid double locking the debouncer mutex `notify()` could return whether the token is debounced